pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -298,61 +375,73 @@ func (fs *Decomposedfs) createHiddenSpaceFolder(ctx context.Context, r *node.Nod
 	return fs.tp.CreateDir(ctx, hiddenSpace)
 }
 
-func (fs *Decomposedfs) createStorageSpace(ctx context.Context, spaceType, nodeID string) error {
+func (fs *Decomposedfs) createStorageSpace(ctx context.Context, spaceType, spaceID string) error {
 	// create space type dir
 	if err := os.MkdirAll(filepath.Join(fs.o.Root, "spaces", spaceType), 0700); err != nil {
 		return err
 	}
 
 	// we can reuse the node id as the space id
-	err := os.Symlink("../../nodes/"+nodeID, filepath.Join(fs.o.Root, "spaces", spaceType, nodeID))
+	err := os.Symlink("../../nodes/"+spaceID, filepath.Join(fs.o.Root, "spaces", spaceType, spaceID))
 	if err != nil {
 		if isAlreadyExists(err) {
-			appctx.GetLogger(ctx).Debug().Err(err).Str("node", nodeID).Str("spacetype", spaceType).Msg("symlink already exists")
+			appctx.GetLogger(ctx).Debug().Err(err).Str("space", spaceID).Str("spacetype", spaceType).Msg("symlink already exists")
 		} else {
 			// TODO how should we handle error cases here?
-			appctx.GetLogger(ctx).Error().Err(err).Str("node", nodeID).Str("spacetype", spaceType).Msg("could not create symlink")
+			appctx.GetLogger(ctx).Error().Err(err).Str("space", spaceID).Str("spacetype", spaceType).Msg("could not create symlink")
 		}
 	}
 
 	return nil
 }
 
-func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, node *node.Node, nodePath, spaceType string, permissions map[string]struct{}) (*provider.StorageSpace, error) {
-	owner, err := node.Owner()
+func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, n *node.Node, nodePath, spaceType string, permissions map[string]struct{}) (*provider.StorageSpace, error) {
+	owner, err := n.Owner()
 	if err != nil {
 		return nil, err
 	}
 
 	// TODO apply more filters
 
-	sname, err := xattr.Get(node.InternalPath(), xattrs.SpaceNameAttr)
-	if err != nil {
+	sname := ""
+	if bytes, err := xattr.Get(n.InternalPath(), xattrs.SpaceNameAttr); err == nil {
+		sname = string(bytes)
+	}
+
+	if err := n.FindStorageSpaceRoot(); err != nil {
 		return nil, err
 	}
+
 	space := &provider.StorageSpace{
-		// FIXME the driver should know its id move setting the spaceid from the storage provider to the drivers
-		//Id: &provider.StorageSpaceId{OpaqueId: "1284d238-aa92-42ce-bdc4-0b0000009157!" + n.ID},
+		Id: &provider.StorageSpaceId{OpaqueId: n.SpaceRoot.ID},
 		Root: &provider.ResourceId{
-			// FIXME the driver should know its id move setting the spaceid from the storage provider to the drivers
-			//StorageId: "1284d238-aa92-42ce-bdc4-0b0000009157",
-			OpaqueId: node.ID,
+			StorageId: n.SpaceRoot.ID,
+			OpaqueId:  n.ID,
 		},
-		Name:      string(sname),
+		Name:      sname,
 		SpaceType: spaceType,
 		// Mtime is set either as node.tmtime or as fi.mtime below
 	}
 
-	user := ctxpkg.ContextMustGetUser(ctx)
-
 	// filter out spaces user cannot access (currently based on stat permission)
```

Comment:
clutter. remove these comments

Reply:
Done in https://github.com/cs3org/reva/pull/2383